### DISCUSSION TITLE
IE11 (blank page - syntax error): Babel-preset-env and webpack config - app not working

### DISCUSSION THREAD

**Discussion by hugoalexandremf** [babel/babel#14368] [Q&A]
I have a React app that until now was working fine in IE11 with @babel/polyfill. Due to the deprecation of this package I removed this dependency and installed `core-js` and `regenerator-runtime`. I configured `babel-preset-env `and added to the `webpack.config` the line `target: ['web', 'es5'],`. 

But now the app only displays a blank page when opened in IE11 - due to syntax error - I can the see the error happens because of this arrow function present in the bundle `sort(((e,t)=>Number(e)`. I found more occurrences of arrow functions in the final bundle.

Could you please point me in the right direction to solve this problem? 

**UPDATE**: I opened a discussion in webpack because I noted that @babel/polyfill is not working with webpack 5.

My `webpack.prod.js` is as follows:
```JSX
var path = require('path');

const TerserPlugin = require("terser-webpack-plugin");
const optimizationOptions = {
     minimize: true,
     minimizer: [
          new TerserPlugin({
               terserOptions: {
                    format: {
                         comments: false,
                    },
               },
               extractComments: false,
          }),
     ],
};

module.exports = [
     {
          mode: 'production',
          target: ['web', 'es5'],
          entry: ['./src/main/resources/templates/js/appIndex.js'],
          output: {
               path: path.resolve(__dirname, "./src/main/resources/static/scripts/"),
               filename: "appScript.js"
          },
          resolve: {
               extensions: ['.js', '.jsx']
          },
          module: {
               rules: [
                    {
                         test: /\.(js|jsx)$/,
                         exclude: /(node_modules)/,
                         use: {
                              loader: 'babel-loader',
                              options: {
                                   presets: [["@babel/preset-env", {
                                        useBuiltIns: "usage",
                                        corejs: 3,
                                        //https://babeljs.io/docs/en/options#targets
                                        //When no targets are specified: Babel will assume you are targeting the oldest browsers possible                                        
                                        targets: {
                                             "ie": "11"
                                        }
                                   }], "@babel/react"]
                              }
                         }
                    },
                    {
                         test: /\.css$/,
                         use: ["style-loader", "css-loader"]
                    }
               ]
          },
          optimization: optimizationOptions
     }
];
```

**Comment by nicolo-ribaudo** [comment]
You are excluding `node_modules` from Babel: is it possible that those `=>` come from one of your dependencies?

**Reply by hugoalexandremf** [reply]
I got the config working with this update (from here: [https://github.com/zloirock/core-js/issues/514#issuecomment-523524472](https://github.com/zloirock/core-js/issues/514#issuecomment-523524472) and here [https://stackoverflow.com/questions/63880144/script5022-exception-thrown-and-not-caught):](https://stackoverflow.com/questions/63880144/script5022-exception-thrown-and-not-caught)

`exclude: /@babel(?:\/|\\{1,2})runtime|core-js/,`

**Reply by nicolo-ribaudo** [reply]
Ok, those dependencies don't include arrows. If you are only excluding them, then I don't know where the arrows came from :thinking: 

It looks like for some reason Babel is not compiling something, but in order to be able to help you I need an example that reproduces the problem.

**Reply by nicolo-ribaudo** [reply]
Oh wait, with that config you fixed the problem?

**Reply by hugoalexandremf** [reply]
Thanks for the reply.
Yes, with this little change in the exclude field the app works in IE11 but I'm still trying to understand why when I exclude node_modules the arrow functions are present in the bundle.
I can put together an example to check specifically what is failing.

**Reply by nicolo-ribaudo** [reply]
It's because one of your dependencies uses `=>`: if you don't compile it, its arrow functions end up being included in your final bundle.
